packages/terra-date-picker/src/DateUtil.jsx

```diff
@@ -142,6 +143,113 @@ class DateUtil {
     return undefined;
   }
 
+  /**
+   * Returns an easily parsable date format string.
+   * @param {string} locale - The locale variant from dateFormatOrder.
+   */
+  static getDateFormat(locale) {
+    let dateVariant;
+
+    if (locale === '0') {
+      dateVariant = 'MM-DD-YYYY';
+    } else if (locale === '1') {
+      dateVariant = 'YYYY-MM-DD';
+    } else {
+      dateVariant = 'DD-MM-YYYY';
+    }
+
+    return dateVariant;
+  }
+
+  /**
+   * Returns an object consisting of placeholder strings and a delimiter,
+   * based on the date format.
+   * @param {string} variant - Variant string obtained from getDateFormat.
+   * @param {string} placeholder - Passed on placeholder prop string.
+   */
+  static getPlaceholderValues(variant, placeholder) {
+    let dateMonth;
+    let dateDay;
+    let dateYear;
+    let dateDelimiter;
+    let ph = placeholder;
+
+    if (!ph) {
+      ph = 'MM-DD-YYYY';
+    }
+
+    if (variant === 'MM-DD-YYYY') {
```

Comment:
Can we drop the variant logic and use the translation to parse the month, day, year?

Reply:
Parsing the translation seems like it would have variations to actually parse? I can get rid of the variant logic, and just the locale format from the date format order to cut some stuff down. That said, if there would be a relatively concise way to parse all of the date translations, I'd be happy to go that route!